api/lib/domain/usecases/generate-organization-learners-username-and-temporary-password.js

```diff
@@ -0,0 +1,165 @@
+import { OrganizationLearnerIdentity } from '../../../src/identity-access-management/domain/models/OrganizationLearnerIdentity.js';
+import { UserNotAuthorizedToUpdatePasswordError } from '../../../src/shared/domain/errors.js';
+import { OrganizationLearnerPasswordResetDTO } from '../../../src/shared/domain/models/OrganizationLearnerPasswordResetDTO.js';
+import {
+  ORGANIZATION_LEARNER_DOES_NOT_BELONG_TO_ORGANIZATION_CODE,
+  ORGANIZATION_LEARNER_WITHOUT_USERNAME_CODE,
+} from '../constants/generate-organization-learners-username-and-temporary-password-errors.js';
+
+const generateOrganizationLearnersUsernameAndTemporaryPassword = async function ({
+  organizationId,
+  organizationLearnersId,
+  userId,
+  cryptoService,
+  passwordGenerator,
+  userReconciliationService,
+  authenticationMethodRepository,
+  organizationRepository,
+  organizationLearnerRepository,
+  userRepository,
+}) {
+  const errorMessage = `User ${userId} cannot reset passwords of some students in organization ${organizationId}`;
+  const organization = await organizationRepository.get(organizationId);
+  const organizationLearners = await organizationLearnerRepository.findByIds({ ids: organizationLearnersId });
+  const userIds = organizationLearners.map((organizationLearner) => organizationLearner.userId);
+  const users = await userRepository.getByIds(userIds);
+  let organizationLearnerIdentities = _buildOrganizationLearnerIdentities({ organizationLearners, users });
+
+  _assertEachOrganizationLearnersBelongToOrganization({
+    errorMessage,
+    organizationId,
+    organizationLearners,
+  });
```

Comment:
Toute cette partie de code pourrait être realisée via un `organizationLearnerIdentity` repository (en SQL) et pourrait s'assurer:
- de faire la jointure entre `user` et `organisation-learners`
- récupérer uniquement les informations nécessaires pour le modèle `OrganizationLearnerIdentity`
- s'assurer que les `organisation-learner` appartiennent bien à l'organisation en ajoutant une clause sql sur l'organisation id

```suggestion
const organization = await organizationRepository.get(organizationId);
let organizationLearnerIdentities = organizationLearnerIdentitiesRepository.findByIds({ ids: organizationId, organizationLearnersId })
```

Reply:
Fait, voir les 3 dernier commits de la PR 😸